src/Sylius/Behat/Context/Setup/OrderContext.php

```diff
@@ -169,10 +187,17 @@ public function customerStartedCheckout(CustomerInterface $customer): void
     public function theCustomerAddedProductToTheCart(CustomerInterface $customer, ProductInterface $product): void
     {
         $cart = $this->createCart($customer);
+
+        /** @var ProductVariantInterface|null $variant */
+        $variant = $this->variantResolver->getVariant($product);
+        if ($variant === null) {
+            throw new \InvalidArgumentException('Product has no variant');
+        }
```

Comment:
```suggestion
        /** @var ProductVariantInterface|null $variant */
        $variant = $this->variantResolver->getVariant($product);
        Assert::notNull($variant, 'Product has no variant');
```
Why assert it here, while all the other places just ignore null;
Better extract it to a private method, `getVariant` or smth, and put the assertion there.
+1 for throwing the exception by itself since it'd ease debugging.